informational/req-res-reliability.md

```diff
@@ -0,0 +1,139 @@
+  - Filter specific failure - we consider service node failing when it cannot serve [subscribe](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md#subscribe) or [ping](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md#subscriber_ping) request with OK status;
+
+## Motivation
+
+Specifications of the mentioned protocols do not define some of the real world use cases that are often observed in unreliable network environment from the perspective of light nodes that are consumers of LightPush and/or Filter protocols.
+Such use cases can be: recovery from offline state, decrease rate of missed messages, increase probability of messages being broadcasted within the network, unreliability of the service node in use.
+
+## Suggestions
+
+### Node health
+
+Node health is a metric meant to determine the connectivity state of a light node and its present ability to reliably send and receive messages from the network.
+We consider this reliability to be dependant on amount of simultaneous connections to responsive service nodes.
+Unfortunately the more connections light node establishes - the more bandwidth is consumed.
+To address this we suggest following states:
+- unhealthy - no connections to service nodes are available regardless of protocol;
+- minimally healthy:
+  - Filter has one service node connection;
+  - LightPush protocol has one service node connection;
+- sufficiently healthy:
+  - Filter has at least 2 connections available to service nodes;
+  - LightPush has at least 2 connections available to service nodes;
+
+### Peers and connection management
+
+#### Pool of reliable service nodes
+Light nodes should maintain a pool of reliable service nodes for each protocol.
+In case service node [fails](./req-res-reliability.md#definitions) to serve protocol request - 
+light node should drop connection to it and a new service node should be connected and added to the pool instead.
+
+We advice to replace service node for LightPush right after first failure in case:
+- connection to it is lost or request timed out;
+- it's response contains [error codes](../standards/core/lightpush.md#examples-of-possible-error-codes): `UNSUPPORTED_PUBSUB_TOPIC`, `INTERNAL_SERVER_ERROR` or `NO_PEERS_TO_RELAY`;
+- request failed but without error message returned;
+
+For Filter we'd recommend replacing service node:
+- [request for subscription](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md#subscribe) so it cannot be initiated;
+- [ping](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md#subscriber_ping) failed 2 times in a row;
```

Comment:
let's address it in following updates to RFC

I haven't done it because we are actually not sure what is a good number 

currently for js-waku it is a [minute](https://github.com/waku-org/js-waku/blob/c0c2907131374a33de441676de1d38a09993aaab/packages/sdk/src/protocols/filter.ts#L48) but technically we can extend it to two minutes or more due to nwaku dropping subscription after 5-10 minutes

if we do so then some other things come into play - what if network is dropped (mitigated by specifying different behavior for it), nwaku changes behavior etc

so, I think let's see how it works with current number and what we see happening to users

Reply:
let's revisit in a month or two